src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,6 +563,9 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
+            
+            if (FileSystem.Current.DirectoryExists(fulldestDirName))
+                throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, fulldestDirName));
```

Comment:
on unix it will still throw if a non-empty directory shows up at the location after the added check. it only doesn't throw when that directory happens to be empty.